### PR TITLE
Versioning docs

### PR DESCRIPTION
**Pull request by jyucsiro**
# Provena PR (Docs): Versioning docs

## Checklist

-   [x] If tests are required for this change, are they implemented?
-   [x] Are user documentation changes required, if so, is there a task to track it and/or is it completed?
-   [x] If migrations are required, is the process documented below?
-   [x] If developer/system documentation updates are required, is there a task to track it and/or is it completed?
-   [ ] At least one developer has reviewed this change (unless PR is being used to mark a commit point without need for review)?
-   [x] If change requires update to the [manual testing runsheet](https://confluence.csiro.au/pages/viewpage.action?pageId=1701138616), is there a task to track it and/or is it completed?

## Description

Adds user facing versioning docs

### CODE REVIEW

File: docs/versioning/versioning-overview.md
Comment (PeterBaker0) [inline docs/versioning/versioning-overview.md]: Can users actually do this, though? At least, does Provena support it?

File: docs/versioning/versioning-overview.md
Comment (PeterBaker0) [inline docs/versioning/versioning-overview.md:42]: Is this important?

File: docs/versioning/versioning-overview.md
Comment (PeterBaker0) [inline docs/versioning/versioning-overview.md:47]: These are not steps - these are axioms i.e. "The versioned item _shall_ refer to the previous version". 

These don't belong in a sequential list, but might still be important?

File: docs/versioning/how-to-version-in-data-store.md
Comment (PeterBaker0) [inline docs/versioning/how-to-version-in-data-store.md:21]: Need to add a catch here for if the user is not on the latest version OR they don't have registry write permissions OR they don't have resource level ADMIN access - all of these will make the button not appear/be greyed out(?).

File: docs/versioning/how-to-version-in-data-store.md
Comment (PeterBaker0) [inline docs/versioning/how-to-version-in-data-store.md:9]: Do we need an explanation here/in the overview/in the FAQ about the DATA in the dataset i.e. the fact that it's a new storage space which is separately managed/accessed.

File: docs/versioning/how-to-version.md
Comment (PeterBaker0) [inline docs/versioning/how-to-version.md]: The VERSIONING tab may not exist - need to add an info/warning block here to explain why it might not be there.

File: docs/versioning/how-to-version.md
Comment (PeterBaker0) [inline docs/versioning/how-to-version.md]: Same as dataset - need the correct permissions + latest version

File: docs/versioning/how-to-version.md
Comment (PeterBaker0) [inline docs/versioning/how-to-version.md:54]: These images are too small 👀

File: docs/versioning/how-to-version.md
Comment (PeterBaker0) [inline docs/versioning/how-to-version.md:48]: Please highlight this in the above screenshot

File: docs/versioning/how-to-version.md
Comment (PeterBaker0) [inline docs/versioning/how-to-version.md]: Cool gif - can we make it bigger

File: docs/versioning/how-to-view-versions.md
Comment (PeterBaker0) [inline docs/versioning/how-to-view-versions.md]: This is important, but only mentioned on this page - its required for every process and maybe not worth stating at this point?

File: docs/versioning/how-to-view-versions.md
Comment (PeterBaker0) [inline docs/versioning/how-to-view-versions.md]: Disclaimer here that this may not appear for non ENTITY types

File: docs/versioning/how-to-view-versions.md
Comment (PeterBaker0) [inline docs/versioning/how-to-view-versions.md:37]: Image width 👁️

File: docs/versioning/how-to-view-versions.md
Comment (PeterBaker0) [inline docs/versioning/how-to-view-versions.md]: Broken link here

File: docs/versioning/how-to-view-versions.md
Comment (PeterBaker0) [inline docs/versioning/how-to-view-versions.md:64]: Do we need this section - maybe just explain that the user can navigate to the Data Store record in the data store, and click on the versioning tab. 

BUT the versioning is also fully functional in the registry for datasets as well - so they could just look at it in either. The view is identical.

**Review by PeterBaker0 — CHANGES_REQUESTED** [review]
Have left comments in various spots. Overall looks good. Have committed some changes to wording where appropriate/not structural. 

In terms of readability, I think that the images should be larger. 

Also, the screenshots with the hand-drawn shapes/arrows imo look a bit amateur/rushed IMO (sorry 😿 )- could we use Greenshot to be consistent (rectangles, numbers, arrows etc).

Example below ;) 
image

In terms of content, I think there a couple of missing details-  highlighted in the review, namely the edge cases aren't covered (versioning not available,  permissions, not on latest version etc). Also, one of the main FAQ points, the data in a dataset not being copied/shared, is missed.

File: docs/versioning/versioning-overview.md
Comment (jyucsiro) [reply docs/versioning/versioning-overview.md]: mm, not yet i think. however, this was some of the original use cases as a first class feature. however, users can still edit the description to say this. maybe we could edit this statement to match.

File: docs/versioning/versioning-overview.md
Comment (jyucsiro) [reply docs/versioning/versioning-overview.md:42]: i think it can be useful... isn't it timestamped? (i think you wrote this in the confluence page originally too)

File: docs/versioning/versioning-overview.md
Comment (jyucsiro) [reply docs/versioning/versioning-overview.md:47]: you're right, this wasn't meant to be steps. they are a list of things that happen when a version is created.

File: docs/versioning/how-to-version-in-data-store.md
Comment (jyucsiro) [reply docs/versioning/how-to-version-in-data-store.md:21]: perhaps we need to add a "pre-requisites" before this

File: docs/versioning/how-to-version-in-data-store.md
Comment (jyucsiro) [reply docs/versioning/how-to-version-in-data-store.md:9]: maybe in the faq and in this page

File: docs/versioning/versioning-overview.md
Comment (fre171csiro) [inline docs/versioning/versioning-overview.md:1]: Would grandma like a diagram to help understand the larger picture?

File: docs/versioning/versioning-overview.md
Comment (fre171csiro) [inline docs/versioning/versioning-overview.md:47]: Point 7. refers to revisions, should this be 'version'

File: docs/versioning/versioning-faq.md
Comment (fre171csiro) [inline docs/versioning/versioning-faq.md:9]: Maybe these could be moved to the main section of FAQs and just a link here to that page

File: docs/versioning/versioning-faq.md
Comment (fre171csiro) [inline docs/versioning/versioning-faq.md:26]: 'Dangling branches' is a very technical term and needs further explanation.  Or removing

File: docs/versioning/versioning-overview.md
Comment (jyucsiro) [reply docs/versioning/versioning-overview.md:1]: Could you suggest a sketch?

File: docs/versioning/versioning-overview.md
Comment (jyucsiro) [reply docs/versioning/versioning-overview.md:47]: yes! you're right.

File: docs/versioning/versioning-faq.md
Comment (fre171csiro) [inline docs/versioning/versioning-faq.md:36]: Is the last point a similarity not a difference?

File: docs/versioning/how-to-view-versions.md
Comment (jyucsiro) [reply docs/versioning/how-to-view-versions.md:64]: I debated about that. I thought I'd do both, as users may have different entry points to versioning. The Data Store view is slightly different - where the buttons are, where to find the lineage feature, etc.. that's all.

File: docs/versioning/how-to-view-versions.md
Comment (jyucsiro) [reply docs/versioning/how-to-view-versions.md]: hmm, maybe we need to hyperlink to the overview section with that list.

File: docs/versioning/how-to-view-versions.md
Comment (jyucsiro) [reply docs/versioning/how-to-view-versions.md]: Suggests we need a Pre-requisites section before each of these how-to's

File: docs/versioning/how-to-version.md
Comment (jyucsiro) [reply docs/versioning/how-to-version.md]: i guess we could remove the fixed width attribute?

File: docs/versioning/how-to-view-versions.md
Comment (fre171csiro) [reply docs/versioning/how-to-view-versions.md:37]: There are smaller images (hard to read) with a width of 400, maybe they should be all 600px wide

File: docs/versioning/how-to-version.md
Comment (fre171csiro) [inline docs/versioning/how-to-version.md:72]: This is cool and probably needs further explanation.  The automation is clear to me what you are illustrating however to a new person its probably just noise.  The other interesting component would be to highlight the reasons for the new versions.

File: docs/versioning/versioning-faq.md
Comment (LindaThomas009) [inline docs/versioning/versioning-faq.md:9]: May need a "why would I need a new version?" so a new user can understand

File: docs/versioning/index.md
Comment (LindaThomas009) [inline docs/versioning/index.md:1]: Might need to check the nav order of the child pages

File: docs/versioning/how-to-view-versions.md
Comment (fre171csiro) [inline docs/versioning/how-to-view-versions.md:99]: This has the wrong file name and is not rendering on the deployed page. I think the correct file is "[../assets/images/versioning/versioning-registry-view-lineage.png](https://github.com/provena/provena/pull/16/files#diff-60b40641e98e28c4fb4e0f21ae07f91fea573d3d1704bd67e01693973190cb3e)"

**Review by fre171csiro — COMMENTED** [review]
Linda and I have reviewed this together.

File: docs/versioning/index.md
Comment (jyucsiro) [reply docs/versioning/index.md:1]: What would be the suggested order?

Comment (LindaThomas009) [reply docs/versioning/index.md:1]: (Some of the pages had the same nav order number)
How about:
- versioning-overview
- how-to-version-in-data-store
- how-to-version (registry)
- how-to-view-versions

If the FAQs get moved to the main FAQ page, then a link at the bottom of each page to the FAQs would be a good idea

Comment (jyucsiro) [reply docs/versioning/index.md:1]: Hows this? 
image

Comment (jyucsiro) [reply docs/versioning/index.md:1]: Now 
image

File: docs/versioning/versioning-overview.md
Comment (jyucsiro) [reply docs/versioning/versioning-overview.md]: edited the statement now

File: docs/versioning/how-to-version-in-data-store.md
Comment (jyucsiro) [reply docs/versioning/how-to-version-in-data-store.md:9]: updated faq and in this page

File: docs/versioning/how-to-version.md
Comment (jyucsiro) [reply docs/versioning/how-to-version.md:21]: added pre-requisites

File: docs/versioning/how-to-version.md
Comment (jyucsiro) [reply docs/versioning/how-to-version.md]: added pre-requisites and info about needing to be at the latest version

File: docs/versioning/how-to-version.md
Comment (jyucsiro) [reply docs/versioning/how-to-version.md]: removed width restriction

File: docs/versioning/how-to-version.md
Comment (jyucsiro) [reply docs/versioning/how-to-version.md]: done

File: docs/versioning/how-to-version.md
Comment (jyucsiro) [reply docs/versioning/how-to-version.md]: removed

File: docs/versioning/how-to-view-versions.md
Comment (jyucsiro) [reply docs/versioning/how-to-view-versions.md]: done

File: docs/versioning/how-to-view-versions.md
Comment (jyucsiro) [reply docs/versioning/how-to-view-versions.md]: added faq entry

File: docs/versioning/how-to-view-versions.md
Comment (jyucsiro) [reply docs/versioning/how-to-view-versions.md]: fixed

File: docs/versioning/versioning-overview.md
Comment (jyucsiro) [reply docs/versioning/versioning-overview.md:1]: added a screenshot with an image of versioning

File: docs/versioning/versioning-faq.md
Comment (jyucsiro) [reply docs/versioning/versioning-faq.md]: moved

File: docs/versioning/versioning-faq.md
Comment (jyucsiro) [reply docs/versioning/versioning-faq.md]: removed and simplified

File: docs/versioning/versioning-faq.md
Comment (jyucsiro) [reply docs/versioning/versioning-faq.md]: that was confusing. simplified now.

File: docs/versioning/how-to-version.md
Comment (jyucsiro) [reply docs/versioning/how-to-version.md:81]: added commentary on what's happening and some background

File: docs/versioning/versioning-faq.md
Comment (jyucsiro) [reply docs/versioning/versioning-faq.md]: added in central faq

**Comment by jyucsiro** [comment]
Thanks all. I have gone through and addressed the comments - would be good to get final review/approval before merging.

File: docs/versioning/index.md
Comment (LindaThomas009) [reply docs/versioning/index.md:1]: Looks good 👍

File: docs/faq.md
Comment (PeterBaker0) [inline docs/faq.md:91]: Something weird going on here

Comment (PeterBaker0) [reply docs/faq.md:91]: And dangling branches still there

File: docs/faq.md
Comment (PeterBaker0) [inline docs/faq.md:101]: This list feels a bit short. 

The new versioned item also has a version number which is incremented and displayed, it links to the previous in the metadata (not just the Version activity).

Comment (PeterBaker0) [reply docs/faq.md:101]: Could just get rid of the list if there's not enough in ther.

File: docs/faq.md
Comment (PeterBaker0) [inline docs/faq.md:116]: hyperlinks here would be nice

logged in -> logging in

Admin permissions -> registry permissions

File: docs/versioning/how-to-version.md
Comment (PeterBaker0) [inline docs/versioning/how-to-version.md:94]: Duplicated

**Review by PeterBaker0 — APPROVED** [review]
A few minor things but looks great. 

Happy to merge.

**Pull request merged**
